src/main/java/com/yahoo/bullet/kafka/CertRefreshingSSLEngineFactory.java

```diff
@@ -0,0 +1,239 @@
+        sslEngine.setSSLParameters(sslParams);
+
+        return sslEngine;
+    }
+
+    @Override
+    public SSLEngine createServerSslEngine(String peerHost, int peerPort) {
+        throw new RuntimeException("This only supports client-side SSLEngines, it does not support createServerSslEngine().");
+    }
+
+    @Override
+    public boolean shouldBeRebuilt(Map<String, Object> nextConfigs) {
+        // This is only used for server-side SslEngineFactories, so we can just return false here on the client-side.
+        // Also, even if it's called on the client-side in future kafka-client versions, we cannot allow this
+        // class to be rebuilt at the SslChannelBuilder / SslFactory level because then the settings
+        // (e.g. keystore/truststore paths) in the KeyManager and TrustManager (which live in the SSLEngine) could get
+        // out-of-sync with the settings in the re-built instance of this class that is being used in the
+        // SslChannelBuilder / SslFactory
+        return false;
+    }
+
+    @Override
+    public Set<String> reconfigurableConfigs() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public KeyStore keystore() {
+        try {
+            return createKeyStore(publicCertLocation, privateKeyLocation);
+        } catch (Exception e) {
+            throw new RuntimeException("Error creating keystore.", e);
+        }
+    }
+
+    @Override
+    public KeyStore truststore() {
+        try {
+            return getKeyStore(truststoreLocation, truststorePassword.value().toCharArray());
+        } catch (Exception e) {
+            throw new RuntimeException("Error creating truststore.", e);
+        }
+    }
+
+    @Override
+    public void close() throws IOException {
+        this.sslContext = null;
+    }
+
+    // Functions that proxy to other classes or require real certs can be extended for testing
+
+    /**
+     * Create an SSLContext.
+     *
+     * @return the {@link SSLContext}.
+     */
+    protected SSLContext createSSLContext() {
+        try {
+            KeyRefresher keyRefresher = generateKeyRefresher(truststoreLocation,
+                                                             truststorePassword.value(),
+                                                             publicCertLocation,
+                                                             privateKeyLocation);
+            SSLContext sslContext = Utils.buildSSLContext(keyRefresher.getKeyManagerProxy(), keyRefresher.getTrustManagerProxy());
+            keyRefresher.startup(keyRefreshInterval);
+            log.info("Creating SSLContext that will refresh keys every {} ms.", keyRefreshInterval);
+            return sslContext;
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to create SSLContext.", e);
+        }
+    }
+
+    /**
+     * Generate a keystore from the given jks file and password.
+     *
+     * @param jksFilePath the path to the jks file.
```

Comment:
Capitalize the first letter of the sentence for the javadoc params and the return here and below